src/pid/sd-jwt/index.ts

```diff
@@ -0,0 +1,108 @@
+import base64decode from "../lib/base64";
+import { PID, Disclosure, SdJwt4VC } from "./types";
+import { pidFromToken } from "./converters";
+
+/**
+ * Decode a given SD-JWT with Disclosures to get the parsed PID object they define.
+ * It ensures provided data is in a valid shape.
+ *
+ * It DOES NOT verify token signature nor check disclosures are correctly referenced by the SD-JWT.
+ * Use {@link verify} instead
+ *
+ * @function
+ * @param token The encoded token that represents a valid sd-jwt for verifiable credentials
+ *
+ * @returns The validated PID object along with the parsed SD-JWT token and the parsed disclosures
+ * @throws A decoding error if the token doesn't resolve in a valid SD-JWT
+ * @throws A validation error if the provided data doesn't result in a valid PID
+ *
+ */
+export function decode(token: string): PidWithToken {
+  // token are expected in the form "sd-jwt~disclosure0~disclosure1~...~disclosureN"
+  const [rawSdJwt = "", ...rawDisclosures] = token.split("~");
+
+  // get the sd-jwt as object
+  // validate it's a valid SD-JWT for Verifiable Credentials
+  const [header, payload] = rawSdJwt
+    .split(".")
+    .slice(0, 2)
+    .map(base64decode)
+    .map((e) => JSON.parse(e));
```

Comment:
ok for now. This part must then be replaced with https://github.com/pagopa/io-react-native-jwt

Reply:
done in #7